client/Sample/tf_freerdp.c

```diff
@@ -338,12 +338,13 @@ int main(int argc, char* argv[])
 		goto fail;
 
 	status = freerdp_client_settings_parse_command_line(context->settings, argc, argv, FALSE);
-	status =
-	    freerdp_client_settings_command_line_status_print(context->settings, status, argc, argv);
-
 	if (status)
 	{
-		rc = 0;
+		freerdp_client_settings_command_line_status_print(context->settings, status, argc, argv);
```

Comment:
Since this is always the same code, it might be a good idea to integrate this check in `freerdp_client_settings_command_line_status_print` and return the desired exit code, so you just need `rc = freerdp_client_settings_command_line_status_print(context->settings, status, argc, argv);`

Reply:
I initially made this the way you suggests and then I have changed it to the current version to not break unofficial clients...